### PR TITLE
Added an FAQ entry about MPI hang issue.

### PR DESCRIPTION
**Pull request by keisukefukuda**
This PR adds an entry to the FAQ section about MPI process hang issue.

### CODE REVIEW

**Comment by keisukefukuda** [comment]
Generated HTML on my environment:

image

File: docs/source/tutorial/tips_faqs.rst [hunk 1]
```diff
@@ -39,3 +39,76 @@ Please refer to `MPI4py API reference <http://pythonhosted.org/mpi4py/apiref/mpi
...
+    mpi4py.MPI.COMM_WORLD.Barrier()
+
+  if __name__ == '__main__':
+    func()
+
+  # mpiexec -n 2 python test.py
+
+To avoid this problem, you can inject the following code snippet into your script
+file to force all processes to abort if an uncaught exception occurs.::
+
```
Comment (shu65) [inline docs/source/tutorial/tips_faqs.rst]: I think that it is better to make it easier to understand the start of code snipet.

File: docs/source/tutorial/tips_faqs.rst [hunk 2]
```diff
@@ -39,3 +39,76 @@ Please refer to `MPI4py API reference <http://pythonhosted.org/mpi4py/apiref/mpi
...
+        try:
+            import mpi4py.MPI
+            mpi4py.MPI.COMM_WORLD.Abort(1)
+        except Exception as e:
+            sys.stderr.write("Sorry, we failed to stop MPI, this MPI process may hang.\n")
+            sys.stderr.flush()
+            raise e
+
+  sys.excepthook = global_except_hook
+
```
Comment (shu65) [inline docs/source/tutorial/tips_faqs.rst]: I think that it is better to make it easier to understand the end of code snipet.

File: docs/source/tutorial/tips_faqs.rst (hunk 1 above)
Comment (keisukefukuda) [reply docs/source/tutorial/tips_faqs.rst]: Thanks, I updated the code!

**Comment by keisukefukuda** [comment]
I added two notes:
 * The techniques do not work if a process crashes with `SIGSEGV` or similar low-level signals
 * `-m mpi4py` technique works.

**Pull request merged**